lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -133,19 +133,110 @@ class StableHLOToTTIRTransposeOpConversionPattern
     tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
         srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
 
-    assert(adaptor.getPermutation().size() == 2 &&
-           "TTIR only supports only two dimensional transposeOp.");
+    if (!checkBasicLegality(srcOp, adaptor)) {
+      LLVM_DEBUG(llvm::dbgs() << "Failed legality checks in the handling of Op "
+                              << srcOp->getName());
+      return failure();
+    }
 
     rewriter.replaceOpWithNewOp<mlir::tt::ttir::TransposeOp>(
         srcOp, outputTensor.getType(), Value(adaptor.getOperand()),
         Value(outputTensor), adaptor.getPermutation()[0],
         adaptor.getPermutation()[1],
+
```

Comment:
Just curious, is the clang formatter inserting this empty line for you or..?

Reply:
Yes something weird is happening, manually deleting now.